simpleclient_pushgateway/src/test/java/io/prometheus/client/exporter/PushGatewayTest.java

```diff
@@ -36,6 +36,20 @@ public void setUp() {
     groupingKey.put("l", "v");
   }
 
+  @Test(expected = RuntimeException.class)
+  public void testInvalidURLThrowsRuntimeException() {
+    new PushGateway("::"); // ":" is interpreted as port number, so parsing fails
+  }
+
+  @Test
+  public void testMultipleSlashesAreStrippedFromURL() {
+    final PushGateway pushGateway = new PushGateway("example.com:1234/context///path//");
+    Assert.assertEquals(
+        "http://example.com:1234/context/path/metrics/job/",
+        pushGateway.gatewayBaseURL
+    );
```

Comment:
Here's why I've exposed the `gatewayBaseURL` field: it's mostly there to check that the code allows passing an address with a trailing slash to the original constructor, which is new (and improved, in my opinion) behaviour.